src/goto-harness/function_call_harness_generator.cpp

```diff
@@ -48,6 +48,9 @@ struct function_call_harness_generatort::implt
   std::set<irep_idt> function_parameters_to_treat_as_arrays;
   std::set<irep_idt> function_arguments_to_treat_as_arrays;
 
+  std::set<irep_idt> function_parameters_to_treat_as_cstrings;
```

Comment:
does it need to be ordered?

Reply:
there is no particular reason to prefer either ordered or unordered in this case.

Reply:
The preference should be to go for unordered when ordered is not needed (the standard library naming here is not very good in my opinion, set should have been the unordered one, and the ordered one should have been `ordered_set`). Using `std::set` implicitly says that the order matters and unordered set are normally more efficient.

Reply:
+1 to what @romainbrenguier said (and @LAJW beforehand): although I completely agree that the difference here will not be measurable (and may well be worse with `unordered_set`!), across the codebase we use `unordered_*` unless ordering matters. Most of the time it wouldn't matter as a lot of our sets will be small, but consistency just helps. You've tripped up three reviewers (count me in) on such a minor issue. (We're all very good at bike-shedding, of course.)